src/System.Net.NetworkInformation/tests/FunctionalTests/IPInterfacePropertiesTest_OSX.cs

```diff
@@ -0,0 +1,168 @@
+
+                    Assert.NotNull(uni.IPv4Mask);
+                    _log.WriteLine("--- IPv4 Mask: " + uni.IPv4Mask);
+                    Assert.Throws<PlatformNotSupportedException>(() => uni.IsDnsEligible);
+                    Assert.Throws<PlatformNotSupportedException>(() => uni.IsTransient);
+                    Assert.Throws<PlatformNotSupportedException>(() => uni.PrefixOrigin);
+                    Assert.Throws<PlatformNotSupportedException>(() => uni.SuffixOrigin);
+
+                    // Prefix Length
+                    Assert.Throws<PlatformNotSupportedException>(() => uni.PrefixLength);
+                }
+
+                Assert.Throws<PlatformNotSupportedException>(() => ipProperties.WinsServersAddresses);
+            }
+        }
+
+        [Fact]
+        public void IPInfoTest_AccessAllIPv4Properties_NoErrors()
+        {
+            foreach (NetworkInterface nic in NetworkInterface.GetAllNetworkInterfaces())
+            {
+                _log.WriteLine("Nic: " + nic.Name);
+
+                IPInterfaceProperties ipProperties = nic.GetIPProperties();
+
+                _log.WriteLine("IPv4 Properties:");
+
+                IPv4InterfaceProperties ipv4Properties = ipProperties.GetIPv4Properties();
+
+                _log.WriteLine("Index: " + ipv4Properties.Index);
+                Assert.Throws<PlatformNotSupportedException>(() => ipv4Properties.IsAutomaticPrivateAddressingActive);
+                Assert.Throws<PlatformNotSupportedException>(() => ipv4Properties.IsAutomaticPrivateAddressingEnabled);
+                Assert.Throws<PlatformNotSupportedException>(() => ipv4Properties.IsDhcpEnabled);
+                Assert.Throws<PlatformNotSupportedException>(() => ipv4Properties.IsForwardingEnabled);
+                _log.WriteLine("Mtu: " + ipv4Properties.Mtu);
+                Assert.Throws<PlatformNotSupportedException>(() => ipv4Properties.UsesWins);
+            }
+        }
+
+        [Fact]
+        public void IPInfoTest_AccessAllIPv6Properties_NoErrors()
+        {
+            foreach (NetworkInterface nic in NetworkInterface.GetAllNetworkInterfaces())
+            {
+                _log.WriteLine("Nic: " + nic.Name);
+
+                IPInterfaceProperties ipProperties = nic.GetIPProperties();
+
+                _log.WriteLine("IPv6 Properties:");
+
+                IPv6InterfaceProperties ipv6Properties = ipProperties.GetIPv6Properties();
+
+                if (ipv6Properties == null)
+                {
+                    _log.WriteLine("IPv6Properties is null");
+                    continue;
+                }
+
+                _log.WriteLine("Index: " + ipv6Properties.Index);
+                _log.WriteLine("Mtu: " + ipv6Properties.Mtu);
+                Assert.Throws<PlatformNotSupportedException>(() => ipv6Properties.GetScopeId(ScopeLevel.Link));
+            }
+        }
+
+        [Fact]
+        [Trait("IPv6", "true")]
```

Comment:
Not sure, this code is grabbed from the same Windows test case. I think @CIPop added them for some easy classification / filtering.